ntp-proto/src/packet/mod.rs

```diff
@@ -858,7 +858,25 @@ impl<'a> NtpPacket<'a> {
                 mac: None,
             },
             #[cfg(feature = "ntpv5")]
-            NtpHeader::V5(_header) => todo!("NTPv5 does not have KISS codes yet"),
+            NtpHeader::V5(header) => NtpPacket {
+                header: NtpHeader::V5(v5::NtpHeaderV5::rate_limit_response(header)),
+                efdata: ExtensionFieldData {
+                    authenticated: vec![],
+                    encrypted: vec![],
+                    // Ignore encrypted so as not to accidentaly leak anything
+                    untrusted: packet_from_client
+                        .efdata
+                        .untrusted
+                        .into_iter()
+                        .chain(packet_from_client.efdata.authenticated)
+                        .filter(|ef| matches!(ef, ExtensionField::UniqueIdentifier(_)))
+                        .chain(std::iter::once(ExtensionField::DraftIdentification(
+                            Cow::Borrowed(v5::DRAFT_VERSION),
+                        )))
+                        .collect(),
```

Comment:
This section now gets repeated a lot. Maybe move it into a function?

Reply:
This may be a good idea, but I suggest we do that in a separate PR for both the NTPv4 and v5 code in one go.